src/Middleware/RequestThrottling/src/Internal/RequestQueue.cs

```diff
@@ -1,73 +0,0 @@
-// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
-
-using System;
-using System.Threading;
-using System.Threading.Tasks;
-
-namespace Microsoft.AspNetCore.RequestThrottling.Internal
-{
-    internal class RequestQueue : IDisposable
```

Comment:
Please keep the RequestQueue and the middleware segregated.

Reply:
My problem was that I wanted to forget about layers and focus on eliminating redundancy. I am re-prioritizing these for future decisions.